crates/ruff/src/checkers/tokens.rs

```diff
@@ -45,12 +45,15 @@ pub(crate) fn check_tokens(
         let mut state_machine = StateMachine::default();
         for &(ref tok, range) in tokens.iter().flatten() {
             let is_docstring = state_machine.consume(tok);
```

Comment:
Does this "state machine" need any updating to correctly handle fstrings?

Reply:
No, as f-strings cannot be used as docstrings. The state machine is mainly used for docstring detection.

Reply:
That's correct. But does it need any updating to not get confused by the new tokens emitted by the lexer?

Reply:
No, it doesn't. The reason being that the detection mechanism resets the state if it finds any other token other than the string token after a function / class / module.